.github/workflows/tests.yml

```diff
@@ -195,6 +195,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Install qemu (required for cross-platform builds)
+        run: |
+          sudo apt update
+          sudo apt -y install qemu-system qemu-user-static
```

Comment:
there is an action for this: https://github.com/docker/setup-qemu-action

Reply:
I added this because it's from docker's official repo. However I agree that we should use minimal of these actions. I also see this is how we install it in other repo workflows. I'm reverting it now.

Reply:
> we limit our ability to reproduce problems we see in CI.

Agreed, although `apt-get` is not going to work on darwin 😄 👀 
Although there are also ways to run github actions locally, i.e. https://github.com/nektos/act